test/integration/graql/query/NumberCastingIT.java

```diff
@@ -128,31 +132,38 @@ public void whenAddressingDateAsLocalDate_ConversionHappens() {
     @Test
     public void whenAddressingDateAsLocalDateTime_ConversionHappens() {
         LocalDateTime now = LocalDateTime.now();
-
         Pattern pattern = Graql.parsePattern("$x " + now + " isa attr-date;");
         verifyWrite(session, pattern);
         verifyRead(session, pattern);
         cleanup(session, pattern);
     }
 
     @Test
-    public void whenAddressingLongAsConvertibleDouble_ConversionHappens() {
+    public void whenReadingLongAsDouble_ConversionHappens() {
         double value = 10.0;
         Pattern pattern = Graql.var("x").val(value).isa("attr-long");
-        verifyWrite(session, pattern);
+        try (Transaction tx = session.transaction(Transaction.Type.WRITE)) {
+            tx.getAttributeType("attr-long").create(10L);
+            tx.commit();
+        }
         verifyRead(session, pattern);
-        cleanup(session, pattern);
     }
 
-    @Rule
-    public ExpectedException expectedException = ExpectedException.none();
+    @Test
+    public void whenWritingLongAsDouble_throw() {
```

Comment:
Should we maybe leave a note in this file to explicitly state that we have similar tests in BDD, but they do not necessarily cover the nuances of casting in Java?

Reply:
added class comment to address this